server/graphql/v2/query/collection/ExpensesCollectionQuery.ts

```diff
@@ -511,6 +452,69 @@ export const ExpensesCollectionQueryResolver = async (
     }
   }
 
+  if (!isNil(args.chargeHasReceipts)) {
+    where[Op.and].push(
+      { type: ExpenseType.CHARGE },
+      sequelize.where(
+        sequelize.literal(
+          `NOT EXISTS (SELECT id from "ExpenseItems" ei WHERE ei."ExpenseId" = "Expense".id and ei.url IS NULL AND ei."deletedAt" IS NULL)`,
```

Comment:
I'm surprised that this one is expensive. Is it because of the subquery? Should it rather be a join?

Reply:
Not really, I just moved all sub-queries bellow the potential READY_TO_PAY query.